src/components/DatePicker/index.android.js

```diff
@@ -4,60 +4,80 @@ import moment from 'moment';
 import TextInput from '../TextInput';
 import CONST from '../../CONST';
 import {propTypes, defaultProps} from './datepickerPropTypes';
+import DateUtils from '../../libs/DateUtils';
 
 class DatePicker extends React.Component {
     constructor(props) {
         super(props);
-
         this.state = {
             isPickerVisible: false,
+            selectedDate: props.value || props.defaultValue,
         };
-
         this.showPicker = this.showPicker.bind(this);
-        this.raiseDateChange = this.raiseDateChange.bind(this);
+        this.setDate = this.setDate.bind(this);
     }
 
-    /**
-     * @param {Event} event
-     */
-    showPicker(event) {
-        this.setState({isPickerVisible: true});
-        event.preventDefault();
+    componentDidUpdate() {
+        const dateValue = DateUtils.getDateAsText(this.props.value);
+        if (this.props.value === undefined || DateUtils.getDateAsText(this.state.selectedDate) === dateValue) {
+            return;
+        }
+        // eslint-disable-next-line react/no-did-update-set-state
+        this.setState({selectedDate: this.props.value});
+        this.textInput.setNativeProps({text: dateValue});
```

Comment:
Why do we need to add `componentDidUpdate`?

Reply:
This has been done in accordance to make it similar to `baseTextInput` . The value prop has been made optional, so we need to update props using uncontrolled updation. Ref: https://github.com/Expensify/App/pull/7627#discussion_r816952202